From a32379586f2e218baf920f753ca2385116669d83 Mon Sep 17 00:00:00 2001
From: zou cao <cao.zou@windriver.com>
Date: Wed, 22 Feb 2017 19:28:02 +0800
Subject: [PATCH] dts:zynq-7000: reserved-memory 0x00000000 to 0x0080000

for fixing dma halt

Zynq-7000 (UG585) munaul show the dma should not access the address
0x00000000 to 0x0080000, these memory resources is filtered by OCM
and SCU, but these memory hasn't been taken by ocm and scu, we
reserve these memory to workaround for dma halt when dma map these
memory.

Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/boot/dts/zynq-7000.dtsi |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/zynq-7000.dtsi b/arch/arm/boot/dts/zynq-7000.dtsi
index 9f2ea96..0bb320a 100644
--- a/arch/arm/boot/dts/zynq-7000.dtsi
+++ b/arch/arm/boot/dts/zynq-7000.dtsi
@@ -41,6 +41,16 @@
 		};
 	};
 
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		rsvd@00000000 {
+			reg = <0x00000000 0x080000>;
+		};
+	};
+
 	pmu {
 		compatible = "arm,cortex-a9-pmu";
 		interrupts = <0 5 4>, <0 6 4>;
-- 
1.7.5.4

